float64 width
float64 speed
bool enable
